RLBotDiscordBot/sendclip.py

```diff
@@ -0,0 +1,77 @@
+import json
+import logging
+import requests
+import sys
+import random as r
+
+import discord
+from discord.ext import commands
+from discord.utils import get
+
+async def sendclip(message):
+    Has_link = False
+    valid_link = False
+    base_channel = 0
+    base_id = 0
+    args = message.content.split(" ")
+    whitelisted_links = ["clips.twitch.tv","www.twitch.tv","gfycat.com","www.youtube.com","i.gyazo.com","i.imgur.com","gyazo.com","streamable.com","www.gifyourgame.com","cdn.discordapp.com"]
+    for i in range(len(args)):
+        #print(i)
+        #print(args[i])
+        if args[i] == "!sendclip":
+            base_channel = message.channel.id
+            base_id = message.id
+            #print(base_channel)
+            #print(base_id)
+            #print("has sendclip")
+            if len(message.attachments) >= 1:
+                for attachment in message.attachments:
+                    args.append(attachment.url)
+                    #print(attachment.url)
+            del args[i]
+            for j in range(len(args)+1):
+                try:
+                    link_test = args[j].split("://")

```

Comment:
You could use this to write less code I bet: https://docs.python.org/3/library/urllib.parse.html